internal/controller/driver_controller.go

```diff
@@ -981,7 +982,41 @@ func (r *driverReconcile) reconcileNodePluginDeamonSetForCsiAddons() error {
 
 	log := r.log.WithValues("csiAddonsDaemonSetName", daemonSet.Name)
 
-	if !ptr.Deref(r.driver.Spec.DeployCsiAddons, false) {
+	// FIXME: Spec.DeployCsiAddons defaults to true, causes CSI-Addons for CephFS enabled by default
+	withCsiAddonsDaemonSet := false // ptr.Deref(r.driver.Spec.DeployCsiAddons, false)
+	withCsiAddonsVolumeCondition := false
+
+	// FIXME: this workaround allows a user to disable the DaemonSet for RBD
+	if r.isRbdDriver() {
+		withCsiAddonsDaemonSet = ptr.Deref(r.driver.Spec.DeployCsiAddons, false)
+	}
+
+	// check if the driver wants CSI-Addons features
+	if feature := r.driver.GetAnnotations()[driverCSIAddonsFeatureVolumeCondition]; feature != "" {
+		enabled, err := strconv.ParseBool(feature)
+		if err != nil {
+			r.log.Error(
+				err,
+				"Unable to parse annotation on driver.csi.ceph.io",
+				"name",
+				client.ObjectKeyFromObject(&r.driver),
+				driverCSIAddonsFeatureVolumeCondition,
+				feature,
+			)
+			return err
+		}
+
+		withCsiAddonsVolumeCondition = enabled
+		// if the feature is enabled, enable the daemonset too
+		withCsiAddonsDaemonSet = withCsiAddonsDaemonSet || withCsiAddonsVolumeCondition
+	}
+
+	// FIXME: explicitly disable for NFS, depends on #284
```

Comment:
Remove FiXME as we dont want to deploy this for NFS thats why we have this code?

Reply:
Well, in the future we surely want to be able to enable this for NFS too?

Reply:
We will rework on this with proper API change once its becomes to Beta state.